### ISSUE TITLE
Unattended upgrades still active with cluster.spec.updatePolicy 'external'

### ISSUE THREAD

**Issue by bartpeeters** [kubernetes/kops#12105]
Kops version: v1.19.1
Kubectl version: v1.18.10
Nodes AMI image: 099720109477/ubuntu/images/hvm-ssd/ubuntu-focal-20.04-amd64-server-20201014

The [documentation](https://github.com/kubernetes/kops/blob/master/docs/operations/images.md#security-updates) states that setting `cluster.spec.updatePolicy` to 'external' will disable automatic security updates on Ubuntu.

When checking the logs of the kops-configuration service on the node we see the expected log message:
```
UpdatePolicy is External; skipping installation of unattended-upgrades
```

But in the syslog we still see that unattended upgrades are running daily.

The package `unattended-upgrades` is also installed on our Kubernetes nodes and the systemctl trigger `apt-daily-upgrade.trigger` is enabled.

File `/etc/apt/apt.conf.d/20auto-upgrades` contains:
```
APT::Periodic::Update-Package-Lists "1";
APT::Periodic::Unattended-Upgrade "1";
```

While when not using updatePolicy external, it contains:
```
APT::Periodic::Update-Package-Lists "1";
APT::Periodic::Unattended-Upgrade "1";

APT::Periodic::AutocleanInterval "7";
```
as set by Kops `update_service.go`.

So it seems that the used AMI image has unattended-upgrades enabled by default.

Is this correct? Do newer Ubuntu AMI images also have unattended-upgrades enabled by default?
Reading the documentation, it seemed that just setting the updatePolicy to external should have been enough to disable the unattended-upgrades.

**Comment by olemarkus** [comment]
/kind bug

That particular function seems buggy in general. Unattended Upgrades is indeed installed in the AMI by default and that is the sole reason it is configured. We need to explicitly disable it in that file.

**Comment by olemarkus** [comment]
This was fixed by #12123

/close
